### PR TITLE
feat(Moopa): add presence

### PR DESCRIPTION
**Pull request by DevanAbinaya**
## Description 

- Presence for [Moopa](https://moopa.live)
- Updating info dynamically on different pages
- Having buttons to see/watch currently playing anime/manga
- Have icon fallback to return logo if cover not available
- Have options to select what title language to show

## Acknowledgements
- [x] I read the [Presence Guidelines](https://github.com/PreMiD/Presences/blob/main/.github/CONTRIBUTING.md)
- [x] I linted the code by running `yarn format`
- [x] The PR title follows the repo's [commit conventions](https://github.com/PreMiD/Presences/blob/main/.github/COMMIT_CONVENTION.md)

## Screenshots
<details>
<summary> Proof showing the creation/modification is working as expected </summary>

</details>

### CODE REVIEW

File: websites/M/Moopa/metadata.json [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+		"id": "338295424479789057",
+		"name": "factiven"
+	},
+	"service": "Moopa",
+	"description": {
+		"en": "Moopa is a free Anime and Manga streaming website with AniList integration where you can easily track your progress with a single click of a button."
+	},
+	"url": "moopa.live",
+	"version": "1.0.0",
+	"logo": "https://i.imgur.com/BzZw6aQ.png",
```
Comment (DarkVillager) [inline websites/M/Moopa/metadata.json:13]: Logo's must be 512x512 as per [our guideliness](https://github.com/PreMiD/Presences/blob/main/.github/CONTRIBUTING.md#logo).

File: websites/M/Moopa/presence.ts [hunk 2]
```diff
@@ -0,0 +1,234 @@
+const presence = new Presence({
+		clientId: "1139510267311562842",
+	}),
+	browsingTimestamp = Math.floor(Date.now() / 1000);
+
+const enum Assets {
+	Logo = "https://i.imgur.com/BzZw6aQ.png",
```
Comment (DarkVillager) [inline websites/M/Moopa/presence.ts:7]: Ditto

File: websites/M/Moopa/presence.ts [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+}
+
+presence.on("UpdateData", async () => {
+	const presenceData: PresenceData = {
+			largeImageKey: Assets.Logo,
```
Comment (DarkVillager) [inline websites/M/Moopa/presence.ts:12]: Why repeat it when you can just add it here?
```suggestion
			largeImageKey: Assets.Logo,
			startTimestamp: browsingTimestamp,
```

File: websites/M/Moopa/presence.ts [hunk 4]
```diff
@@ -0,0 +1,234 @@
...
+		},
+		[titleLang] = await Promise.all([presence.getSetting("title")]),
+		{ pathname, href } = document.location;
+
+	if (pathname === "/" || pathname === "/en/") {
+		presenceData.details = "Browsing Homepage";
+		presenceData.startTimestamp = browsingTimestamp;
```
Comment (DarkVillager) [inline websites/M/Moopa/presence.ts:19]:
```suggestion
```

File: websites/M/Moopa/presence.ts [hunk 5]
```diff
@@ -0,0 +1,234 @@
...
+	} else if (pathname.includes("/id/"))
+		presenceData.details = "Browsing Anime/Manga";
+	else if (pathname.includes("/profile/")) {
+		presenceData.details =
+			document.querySelector("h1.font-karla.font-bold.text-2xl.pt-7")
+				?.textContent || "Getting data...";
+		presenceData.state = "Viewing profile";
+	} else if (pathname === "/en/anime/trending/") {
+		presenceData.details = "Browsing trending Anime";
+		presenceData.startTimestamp = browsingTimestamp;
```
Comment (DarkVillager) [inline websites/M/Moopa/presence.ts:29]:
```suggestion
```

File: websites/M/Moopa/presence.ts [hunk 6]
```diff
@@ -0,0 +1,234 @@
...
+	} else if (pathname === "/en/anime/popular/") {
+		presenceData.details = "Browsing popular Anime";
+		presenceData.startTimestamp = browsingTimestamp;
```
Comment (DarkVillager) [inline websites/M/Moopa/presence.ts:32]:
```suggestion
```

**Review by DarkVillager — CHANGES_REQUESTED** [review]

File: websites/M/Moopa/presence.ts [hunk 7]
```diff
@@ -0,0 +1,234 @@
...
+	} else if (pathname.includes("/search/") && pathname.includes("/anime/")) {
+		presenceData.details = "Browsing Anime";
+		presenceData.state = "Searching...";
+		presenceData.smallImageKey = Assets.Search;
+	} else if (pathname.includes("/search/") && pathname.includes("/manga/")) {
+		presenceData.details = "Browsing Manga";
+		presenceData.state = "Searching...";
+		presenceData.smallImageKey = Assets.Search;
+	} else {
+		presence.clearActivity();
```
Comment (theusaf) [inline websites/M/Moopa/presence.ts:228]: Probably unneeded, as the activity is immediately set after this
```suggestion
```

File: websites/M/Moopa/presence.ts [hunk 8]
```diff
@@ -0,0 +1,234 @@
...
+						document
+							.querySelector('meta[name="title"]')
+							?.getAttribute("data-title-native") ||
+						document
+							.querySelector('meta[name="title"]')
+							?.getAttribute("data-title-romaji");
+
+					break;
+				}
+			}
```
Comment (theusaf) [inline websites/M/Moopa/presence.ts:175]: This chunk of code is pretty large and repeated, it could be better as its own function.

**Review by theusaf — CHANGES_REQUESTED** [review]

File: websites/M/Moopa/metadata.json [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+	},
+	"service": "Moopa",
+	"description": {
+		"en": "Moopa is a free Anime and Manga streaming website with AniList integration where you can easily track your progress with a single click of a button."
+	},
+	"url": "moopa.live",
+	"version": "1.0.0",
+	"logo": "https://i.imgur.com/lelaLhp.png",
+	"thumbnail": "https://i.imgur.com/pNmZsH3.png",
+	"color": "#141519",
```
Comment (DarkVillager) [inline websites/M/Moopa/metadata.json:15]: This color is the base color used when something is selected or hovered on
```suggestion
	"color": "#FF7F57",
```

**Review by DarkVillager — CHANGES_REQUESTED** [review]

**Review by theusaf — APPROVED** [review]

**Review by Bas950 — APPROVED** [review]

**Pull request merged**